package/MDAnalysis/analysis/helix_analysis.py

```diff
@@ -339,75 +345,48 @@ class HELANAL(AnalysisBase):
-            multiple selections.
-        ref_axis: array-like of length 3, optional
-            The reference axis used to calculate the tilt of the vector of best fit, 
-            and the local screw angles.
-        flatten_single_helix: bool, optional
-            Whether to flatten results if only one selection is passed.
-        verbose : bool, optional
-            Turn on more logging and debugging.
-
-        Returns
-        -------
-        :class:`~MDAnalysis.analysis.helix_analysis.HELANAL`
-        """
-        try:
-            ss = secondary_structure.simple_mode
-        except AttributeError:
-            raise ValueError('The SecondaryStructure instance must be run '
-                             'before passing to HELANAL')
-
-        helices = secondary_structure.residues[ss == 'Helix']
-        continuous = util.group_consecutive_integers(helices.resindices)
-
-        # must have at least 9 continuous residues for helanal
-        selections = []
-        universe = secondary_structure._universe
-        all_res = universe.residues
-        for resindices in continuous:
-            if len(resindices) >= 9:
-                res = all_res[resindices]
-                sel = 'resid {}:{} and {}'.format(res[0].resid,
-                                                  res[-1].resid,
-                                                  select)
-                selections.append(sel)
-        if not selections:
-            raise ValueError('Could not find any helices with at least '
-                             '9 continuous residues. Consider running '
-                             'HELANAL manually')
-
-        return cls(universe, select=selections, ref_axis=ref_axis,
-                   verbose=verbose, flatten_single_helix=flatten_single_helix)
-
     def __init__(self, universe, select='name CA', ref_axis=[0, 0, 1],
-                 verbose=False, flatten_single_helix=True):
+                 verbose=False, flatten_single_helix=True,
+                 split_residue_sequences=True):
         super(HELANAL, self).__init__(universe.universe.trajectory,
                                       verbose=verbose)
         selections = util.asiterable(select)
-        self.atomgroups = [universe.select_atoms(s) for s in selections]
-        for s, ag in zip(selections, self.atomgroups):
-            ids, counts = np.unique(ag.resindices, return_counts=True)
-            if np.any(counts > 1):
-                dup = ', '.join(map(str, ids[counts > 1]))
-                warnings.warn('Your selection {} includes multiple atoms '
-                              'for residues with these resindices: {}.'
-                              'HELANAL is designed to work on one carbon-alpha '
-                              'per residue.'.format(s, dup))
+        atomgroups = [universe.select_atoms(s) for s in selections]
+        consecutive = []
+        # check that residues are consecutive and long enough sequence
```

Comment:
Added checks and warnings that residue sequences are a) consecutive b) at least 9 residues/atoms long